test/element-utils.spec.js

```diff
@@ -0,0 +1,261 @@
+      var div = document.createElement('div')
+      div.setAttribute('role', 'button')
+      expect(!shouldIgnoreElement(div)).ok()
+    })
+
+    it('accepts link role', function () {
+      var div = document.createElement('div')
+      div.setAttribute('role', 'link')
+      expect(!shouldIgnoreElement(div)).ok()
+    })
+  })
+
+  describe('getElementData', function () {
+    it('gets tag', function () {
+      var div = document.createElement('div')
+      var data = getElementData(div)
+      expect(data.tag === 'div').ok()
+    })
+
+    it('gets tree', function () {
+      var div = createTestElement('div')
+      div.innerHTML = '<div><div><div></div></div></div>'
+      var leaf = div.children[0].children[0].children[0]
+      var data = getElementData(leaf)
+      expect(data.tree === 'body > div > div > div > div').ok()
+    })
+
+    it('gets alt', function () {
+      var a = document.createElement('a')
+      a.setAttribute('alt', 'foobar')
+      var data = getElementData(a)
+      expect(data.alt === 'foobar').ok()
+    })
+
+    it('gets class', function () {
+      var div = document.createElement('div')
+      div.setAttribute('class', 'foo bar baz')
+      var data = getElementData(div)
+      expect(data['class'] === 'foo bar baz').ok()
+    })
+
+    it('gets href', function () {
+      var a = document.createElement('a')
+      a.setAttribute('href', 'https://www.google.com/')
+      var data = getElementData(a)
+      expect(data.href === 'https://www.google.com/').ok()
+    })
+
+    it('gets id', function () {
+      var div = document.createElement('div')
+      div.setAttribute('id', 'foobar')
+      var data = getElementData(div)
+      expect(data.id === 'foobar').ok()
+    })
+
+    it('gets name', function () {
+      var input = document.createElement('input')
+      input.setAttribute('name', 'foobar')
+      var data = getElementData(input)
+      expect(data.name === 'foobar').ok()
+    })
+
+    it('gets role', function () {
+      var div = document.createElement('div')
+      div.setAttribute('role', 'button')
+      var data = getElementData(div)
+      expect(data.role === 'button')
+    })
+
+    it('gets title', function () {
+      var button = document.createElement('button')
+      button.setAttribute('title', 'foobar')
+      var data = getElementData(button)
+      expect(data.title === 'foobar').ok()
+    })
+
+    it('gets type', function () {
+      var input = document.createElement('input')
+      input.setAttribute('type', 'hidden')
+      var data = getElementData(input)
+      expect(data.type === 'hidden').ok()
+    })
+  })
+
+  describe('htmlTreeAsString', function () {
+    it('returns the tree up to five elements deep', function () {
+      var div = document.createElement('div')
+      div.innerHTML = '<div><div><div><div><div></div></div></div></div></div>'
+      var leaf = leafChild(div)
+      expect(htmlTreeAsString(leaf) === 'div > div > div > div > div').ok()
```

Comment:
tiny nit: These tests don't ensure that the ancestry string is in the correct order